WebAPI/Controllers/BooksController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WebAPI.DTOs;
+using WebAPI.Interfaces;
+using Swashbuckle.AspNetCore.Annotations;
+using System.Net;
+using WebAPI.Services;
+using WebAPI.Models;
+using Microsoft.AspNetCore.Authorization;
+
+namespace WebAPI.Controllers
+{
+    [Authorize(Roles = "Admin")]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IGetterService<Book> _getter;
+        private readonly ICreatorService<Book> _creator;
+        private readonly IUpdaterService<Book> _updater;
+        private readonly IDeleterService<Book> _deleter;
+
+        public BooksController(ICreatorService<Book> creator, IGetterService<Book> getter, IUpdaterService<Book> updater, IDeleterService<Book> deleter)
+        {
+            _creator = creator;
+            _getter = getter;
+            _updater = updater;
+            _deleter = deleter;
+        }
```

Comment:
As I told to Justyn already, create some sort of CRUD service that combines all of the getters, updaters etc. This way you won't need to have so many constructor arguments/class fields

Reply:
Don't we have to follow single responsibility principle?